Test/ServerTelemetryChannel.Test/Shared.Tests/Helpers/StubTelemetryBuffer.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Helpers
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Implementation;
+    using System.Threading.Tasks;
+
+    internal class StubTelemetryBuffer : Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Implementation.TelemetryBuffer
+    {
+        public Action<IEnumerable<ITelemetry>> OnSerialize = _ => { };
+
+        public StubTelemetryBuffer(TelemetrySerializer serializer, IApplicationLifecycle applicationLifecycle) : base(serializer, applicationLifecycle)
+        {            
+
+        }
+        public override Task FlushAsync()
+        {
+            // Intentionally blank to simulate situation where buffer is not emptied.
```

Comment:
make it property as for `OnSerialize`. The class intentiallally called `Stub` to be able to override methods